### PR TITLE
kfixes the key issue for filtered delays

### PR DESCRIPTION
**Pull request by alan-stokes**
fixes https://github.com/SpiNNakerManchester/sPyNNaker/issues/850

### CODE REVIEW

**Comment by Christian-B** [comment]
When https://github.com/SpiNNakerManchester/sPyNNaker/issues/850 is fixed correctly this PR is useless.

The whole delay extension vertex should either never have existed or no longer exist.

IE. There should be no core wasted, no DSG/E time wasted for a vertex that has ZERO use!

Suggest closing this NOT MERGED!

**Comment by alan-stokes** [comment]
no. this is needed to ensure that current MASTER delay extensions would never fire IF they got packets. and brings them in line with the rest of the tools with has key api

File: neural_modelling/src/delay_extension/delay_extension.c [hunk 1]
```diff
@@ -156,6 +158,7 @@ static inline uint32_t round_to_next_pot(uint32_t v) {
 static bool read_parameters(struct delay_parameters *params) {
     log_debug("read_parameters: starting");
 
+    has_key = params->has_key;
```
Comment (rowleya) [inline neural_modelling/src/delay_extension/delay_extension.c:161]: I guess this might happen if the filtering removes enough...

**Review by rowleya — APPROVED** [review]
Looks fine to me

**Comment by rowleya** [comment]
I see some previous controversy.  The issue is that we can't currently remove a vertex after edge filtering has happened.  Having a has_key check doesn't take much effort so adding here is fine for now.  We could review has_key in general over the whole application after edge merging is done.  An issue has been added to review this later, but this needs merging to fix master now.

**Pull request merged**